stored_requests/config/config_test.go

```diff
@@ -41,12 +42,87 @@ func isMemoryCacheType(cache stored_requests.CacheJSON) bool {
 }
 
 func TestNewEmptyFetcher(t *testing.T) {
-	fetcher := newFetcher(&config.StoredRequests{}, nil, nil)
-	if fetcher == nil {
-		t.Errorf("The fetcher should be non-nil, even with an empty config.")
+
+	type testCase struct {
+		config       *config.StoredRequests
+		emptyFetcher bool
+		message      string
+	}
+	testCases := []testCase{
+		{
+			config:       &config.StoredRequests{},
+			emptyFetcher: true,
+			message:      "If the config is empty, and EmptyFetcher should be returned",
+		},
+		{
+			config: &config.StoredRequests{
+				Postgres: config.PostgresConfig{
+					CacheInitialization: config.PostgresCacheInitializer{
+						Query: "test query",
+					},
+					PollUpdates: config.PostgresUpdatePolling{
+						Query: "test poll query",
+					},
+					FetcherQueries: config.PostgresFetcherQueries{
+						QueryTemplate: "",
+					},
+				},
+			},
+			emptyFetcher: true,
+			message:      "If Postgres fetcher query is not defined, but Postgres Cache init query and Postgres update polling query are defined EmptyFetcher should be returned",
+		},
+		{
+			config: &config.StoredRequests{
+				Postgres: config.PostgresConfig{
+					CacheInitialization: config.PostgresCacheInitializer{
+						Query: "",
+					},
+					PollUpdates: config.PostgresUpdatePolling{
+						Query: "",
+					},
+					FetcherQueries: config.PostgresFetcherQueries{
+						QueryTemplate: "test fetcher query",
+					},
+				},
+			},
+			emptyFetcher: false,
+			message:      "If Postgres fetcher query is  defined, but Postgres Cache init query and Postgres update polling query are not defined not EmptyFetcher (DBFetcher) should be returned",
+		},
+		{
+			config: &config.StoredRequests{
+				Postgres: config.PostgresConfig{
+					CacheInitialization: config.PostgresCacheInitializer{
+						Query: "test cache query",
+					},
+					PollUpdates: config.PostgresUpdatePolling{
+						Query: "test poll query",
+					},
+					FetcherQueries: config.PostgresFetcherQueries{
+						QueryTemplate: "test fetcher query",
+					},
+				},
+			},
+			emptyFetcher: false,
+			message:      "If Postgres fetcher query is  defined and Postgres Cache init query and Postgres update polling query are  defined not EmptyFetcher (DBFetcher) should be returned",
+		},
 	}
-	if _, ok := fetcher.(empty_fetcher.EmptyFetcher); !ok {
-		t.Errorf("If the config is empty, and EmptyFetcher should be returned")
+
+	for _, test := range testCases {
+
+		fetcher := newFetcher(test.config, nil, &sql.DB{})
+		if fetcher == nil {
+			t.Errorf("The fetcher should be non-nil.")
+		}
+		if test.emptyFetcher {
+			if _, ok := fetcher.(empty_fetcher.EmptyFetcher); !ok {
```

Comment:
Please consider using `assert.Equals` here and in the `else` part of this condition